Add ToggleButtonGroupField (#3808)
vivid-planet/comet: docs/docs/7-migration-guide/migration-from-v7-to-v8.md

  Using import type ensures that types do not introduce unintended runtime dependencies.

</details>

### `FinalFormToggleButtonGroup` deprecated

> `FinalFormToggleButtonGroup` has been deprecated and a new component `ToggleButtonGroupField` got introduced that has the Final Form Field wrapped around it.

Reviewer: ```suggestion
`FinalFormToggleButtonGroup` has been deprecated and a new component `ToggleButtonGroupField` was introduced that includes the Final Form `Field`.
```

> ```

Reviewer: ```suggestion
```diff
```

- import { FinalFormToggleButtonGroup } from "@comet/cms-admin";
+ import { ToggleGroupButtonField } from "@comet/admin";

...
+ FormValueType = "value1" | "value2";

- <Field
-   name="formValue"
-   label={"Field Label"}
-   component={FinalFormToggleButtonGroup}
-   options={[
-       { value: "value1", icon: <Info /> },

> -       { value: "value2", icon: <Info /> },

Reviewer: that values probably should be the same in the diff example

```suggestion
-       { value: "value2", icon: <Error /> },
```

-   ]}
-   optionsPerRow={2}
- />
+ <ToggleGroupButtonField<FormValueType>
+    name="formValue"
+    label={"Field Label"}
+    options={[
+        { value: "value1", label: <Info /> },
+        { value: "value2", label: <Error /> },
+    ]}
+    optionsPerRow={2}
+    />
```

> The `FinalFormToggleButtonGroup` component is still available, but moved from `@comet/cms-admin` to `@comet/admin` package. Furthermore, the prop `options` of the component have been changed from `icon` to `label`.

Reviewer: ```suggestion
The `FinalFormToggleButtonGroup` component is still available, but moved from `@comet/cms-admin` to `@comet/admin` package. Furthermore, the value `icon` in the `options` prop has been renamed to `label`.
```
